quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs

```diff
@@ -173,22 +173,29 @@ async fn mark_splits_as_published_helper(
     if split_ids.is_empty() {
         return Ok(Vec::new());
     }
-    let publishable_states = [SplitState::Staged.as_str(), SplitState::Published.as_str()];
     let published_split_ids: Vec<String> = sqlx::query(
         r#"
         UPDATE splits
-        SET split_state = $1
+        SET
+            split_state = 'Published',
+            -- The values we compare with are *before* the modification:
+            update_timestamp = CASE
+                WHEN split_state = 'Staged' THEN (CURRENT_TIMESTAMP AT TIME ZONE 'UTC')
+                ELSE update_timestamp
+            END,
+            publish_timestamp = CASE
+                WHEN split_state = 'Staged' THEN (CURRENT_TIMESTAMP AT TIME ZONE 'UTC')
+                ELSE publish_timestamp
+            END
         WHERE
-                index_id = $2
-            AND split_id = ANY($3)
-            AND split_state = ANY($4)
+                index_id = $1
+            AND split_id = ANY($2)
+            AND split_state IN ('Published', 'Staged')
```

Comment:
I removed the placeholders for the split states here because we hard code the `Staged` split state within the triggers anyway, and need also hard code it in the `CASE` statement, so to avoid confusion placeholders were removed.